### PR TITLE
eBay (Independent Publisher)

### PR DESCRIPTION
**Pull request by artesianst**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [x] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by Amjed-Ayoub** [comment]
Hello @artesianst,
Thank you so much for your new proposal submission. We are looking forward to seeing your connector artifacts soon.
Please let us know if you have any questions.
Thank you so much again for working with us.

**Comment by artesianst** [comment]
> Hello @artesianst, Thank you so much for your new proposal submission. We are looking forward to seeing your connector artifacts soon. Please let us know if you have any questions. Thank you so much again for working with us.

Thanks for your kind words @Amjed-Ayoub. We will come back with updates soon.

**Comment by artesianst** [comment]
Hello @Amjed-Ayoub,

We are pushing the first version for your review and approval.

Thanks,
Amit.

File: independent-publisher-connectors/eBay/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,6241 @@
...
+            "type": "string",
+            "description": "A unique merchant-defined key (ID) for an inventory location.",
+            "x-ms-summary": "Merchant Location Key",
+            "x-ms-url-encoding": "single"
+          }
+        ]
+      },
+      "post": {
+        "responses": {
+          "default": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/eBay/apiDefinition.swagger.json:3138]: @artesianst it is not allowed to have a default response only. Please define a proper response. please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/coding-standards#responses).

File: independent-publisher-connectors/eBay/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,6241 @@
...
+            "description": "default",
+            "schema": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/eBay/apiDefinition.swagger.json:3140]: @artesianst empty schema is not allowed. please define a proper schema.

File: independent-publisher-connectors/eBay/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,6241 @@
...
+                  "title": "SKU"
+                },
+                "status": {
+                  "type": "string",
+                  "description": "The value specifies the status of the offer.",
+                  "title": "Status"
+                },
+                "storeCategoryNames": {
+                  "type": "array",
+
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/eBay/apiDefinition.swagger.json:6203]: @artesianst can you please explain this empty path. is it possible to remove it please?

File: independent-publisher-connectors/eBay/apiProperties.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
+{
+  "properties": {
+    "connectionParameters": {
+      "token": {
+        "type": "oauthSetting",
+        "oAuthSettings": {
+          "identityProvider": "oauth2",
+          "clientId": "AmitShar-Artesian-SBX-f1c9d9c1e-a4e1f868",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/eBay/apiProperties.json:8]: @artesianst secrets look alike values are not allowed. plese replace with dummy data.

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by artesianst** [comment]
Hello @Amjed-Ayoub, we have fixed as per your suggestions. Please review.

**Comment by Amjed-Ayoub** [comment]
@artesianst thank you so much for fixing your connector files. May I ask you please to provide some screenshots confirming that you tested your connector?

**Comment by Ajinder31** [comment]
Hello @Amjed-Ayoub, My bad. I had added the screenshots in readme file, but it seems not to be working. Please find some screenshot attached below and let me know if anything is missing or require any modifications. Thank you.

[Connector Test Operation](https://astllp-my.sharepoint.com/:i:/g/personal/ajinder_singh_artesian_io/EY0V0ocK9J1AlIPJILXML-gBeTz3pQgXp_XWlaiqzP5yRw?e=mBbQdg)

[eBay Connector](https://astllp-my.sharepoint.com/:i:/g/personal/ajinder_singh_artesian_io/EbHEnhsgRXxInoIk119cRagBuJO0skjOFu5bJhnaPYAXpQ?e=XNwTGa)

[Flow Test 1](https://astllp-my.sharepoint.com/:i:/g/personal/ajinder_singh_artesian_io/EdP7paBBmHhAumjz-3nR2ccB-PhRDfYNKoeJgMzGEWPX_A?e=SL6KAt)

[Flow Test 2](https://astllp-my.sharepoint.com/:i:/g/personal/ajinder_singh_artesian_io/ESHQRmuT39pFhOyqFuMiNVgBiKNFoEgDvcmtQYPVr27O_Q?e=e8hACx)

[Flow Test 3](https://astllp-my.sharepoint.com/:i:/g/personal/ajinder_singh_artesian_io/EcCDvt1WeLlEsVrx6eQjjAoBlaTGHIjZx-I5Rc5opcIkwg?e=YA9IxU)

[Get Inventory Items](https://astllp-my.sharepoint.com/:i:/g/personal/ajinder_singh_artesian_io/ERAvbYVc7tNAjo7JXd53N60BZuclWPzMWzqXcvrWNt18Hg?e=QhTp8J)

[Create Inventory Location](https://astllp-my.sharepoint.com/:i:/g/personal/ajinder_singh_artesian_io/ESwWtixvY-FPrpPC3Qxdq9gB3xmtKJeLzrnvtYKxZFgrRw?e=0Zs3wU)

[Create or Replace Inventory Item](https://astllp-my.sharepoint.com/:i:/g/personal/ajinder_singh_artesian_io/EQ_vPjiit6hHsWR1jrZkEosB94S635ie9-OO4hCnxdgjwg?e=6m3Sgk)

[Get Fulfillment Policy](https://astllp-my.sharepoint.com/:i:/g/personal/ajinder_singh_artesian_io/Eb116GPKY1FNjaO1tUTIvKkB8Ifmj50gy-V_1KvIsNOfWQ?e=xu6kuq)

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @artesianst,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by artesianst** [comment]
Thanks @Amjed-Ayoub for your update and kind words. We will eagerly wait for the production deployment success update.

**Comment by artesianst** [comment]
Thanks for the update @Amjed-Ayoub

**Review by sriyen-msft — APPROVED** [review]

**Comment by sriyen-msft** [comment]
@artesianst Congratulations! The deployment of your connector and documentation page is now complete, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program, thanks.

**Comment by Ajinder31** [comment]
Hello @sriyen-msft  

Thank you for the update.

When we tried to use the eBay connector in PA we got the following error.
eBay_Connector_Error

Please suggest if we're missing anything. Thank you.

@Amjed-Ayoub

**Comment by troystaylor** [comment]
Can confirm the connection doesn't pass client ID. +1 following.

**Comment by Ajinder31** [comment]
Hi @troystaylor, Thank you for the confirmation. May we know if this is due to some misconfiguration in custom connector package or something else?

Also, is it possible to configure connector in such a way that end user be able to fill client Id and client secret associated to their own accounts? If yes, it would be great if you can refer me to the documentation. Thank you.

**Comment by Amjed-Ayoub** [comment]
Hi @Ajinder31 and @troystaylor thank you so much for the confirmation. we will investigate this issue within next week.
Thank you so much again,
Best regards
Amjed Ayoub

**Comment by troystaylor** [comment]
I might suggest this connector gets pulled until the issue is identified and resolved. Thoughts @jopanchal ?

**Comment by Amjed-Ayoub** [comment]
Hello @artesianst,

I hope you are doing well.

Congratulations on completing your "Quickbooks" connector. 

I tried the same approach with "eBay", but I was still getting the same error.
 
image

I also tried creating a new account with eBay and tested the connector with two different accounts but with the same error message.

Can you please confirm if we are using the correct [source](https://developer.ebay.com/api-docs/static/gs_create-the-ebay-api-keysets.html) to get the credentials for this app?

I look forward to hearing from you.

Thank you so much.

**Comment by Amjed-Ayoub** [comment]
@artesianst is it possible to [create an environment](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region) in the preview region to [test the connector](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector)? 
Please let me know so I can deploy the connector to preview region for you to test on your end.
Thank you again.

**Comment by Ajinder31** [comment]
Hi @Amjed-Ayoub, Thank you for the update.

Sure, we will be able to create environment in preview region and test it. Let us know when it deployed to preview region.

**Comment by Ajinder31** [comment]
Hi @Amjed-Ayoub, any updates on deployment to preview environment?

**Comment by Ajinder31** [comment]
Thank you for the update @Amjed-Ayoub. I will verify and update here.

**Comment by Amjed-Ayoub** [comment]
@artesianst this great news. May I ask you please to make the same changes you did in the apiProperties file for Quickbooks connector. and submit a new pull request.

Please let me know if you have any questions.

**Comment by Ajinder31** [comment]
Sure @Amjed-Ayoub. I will take care of it.

**Pull request merged**